projects/dotcms-field-elements/src/components/dot-tags/components/dot-autocomplete/dot-autocomplete.e2e.ts

```diff
@@ -4,197 +4,230 @@ describe('dot-autocomplete', () => {
 
-                expect(await element.find('input')).toBeDefined();
-            });
-
-            it('should not broke when data does is not a function', async () => {
-                element.setAttribute('data', {});
+                input.type('res');
                 await page.waitForChanges();
 
-                expect(await element.find('input')).toBeDefined();
+                const lis = await element.findAll('ul li');
+                expect(lis.length).toBe(3);
             });
         });
+
+        xdescribe('debounce', () => {});
+
+        xdescribe('data', () => {});
     });
 
-    describe('show options', () => {
+    describe('@Events', () => {
         let input;
+        let spySelectEvent;
 
         beforeEach(async () => {
-            await page.$eval('dot-autocomplete', (elm: any) => {
-                elm.data = () => ['tag-1', 'label'];
-            });
-
-            await page.waitForChanges();
-
             input = await page.find('input');
-            await input.press('t');
-            await page.waitForChanges();
+            spySelectEvent = await element.spyOnEvent('select');
         });
 
-        it('should put get data', async () => {
-            const ul = await element.find('ul');
-            expect(ul.innerHTML).toEqualHtml(`
-                <li data-result="tag-1" class="autoComplete_result" tabindex="1">
-                    <span class="autoComplete_highlighted">t</span>ag-1
-                </li>
-            `);
-        });
+        describe('select', () => {
+            it('should trigger when press enter', async () => {
+                await input.type('test');
+                input.press('Enter');
+                await page.waitForChanges();
 
-        it('should clean value and hide options when press esc', async () => {
-            await input.press('Escape');
-            await page.waitForChanges();
+                expect(spySelectEvent).toHaveReceivedEventDetail('test');
+            });
 
-            const ul = await element.find('ul');
-            expect(ul.innerHTML).toEqualHtml('');
-        });
+            it('should trigger when keyboard select a option', async () => {
+                input.type('res');
+                await page.waitForChanges();
 
-        it('should clean options when lost focus', async (done) => {
-            element.spyOnEvent('lostFocus').then(async () => {
+                element.press('ArrowDown');
+                element.press('ArrowDown');
+                element.press('ArrowDown');
+                element.press('Enter');
                 await page.waitForChanges();
-                const options = await page.findAll('li');
-                expect(options.length).toBe(0);
-                done();
-            });
 
-            await input.triggerEvent('blur');
+                expect(spySelectEvent).toHaveReceivedEventDetail('result-3');
+            });
         });
 
+        xdescribe('lostFocus', () => {
```

Comment:
remove?

Reply:
I want to keep it, we need to find a way to tests blur.